### PR TITLE
[DOCS] Adds response body documentation to GET inference API

### PR DESCRIPTION
**Pull request by szabosteve**
This PR expands the GET inference API documentation with the definitions of the response objects.

Preview: http://elasticsearch_53050.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/get-inference.html

### CODE REVIEW

File: docs/reference/ml/ml-shared.asciidoc [hunk 1]
```diff
@@ -1422,6 +1423,131 @@ The number of `partition` field values that were analyzed by the models. This
...
+The time when the trained model was created.
+
+`tags`:::
+(string)
+A comma delimited string of tags. A {infer} model can have many tags, or none.
+
+`metadata`:::
+(object)
+An object containing metadata about the trained model. Includes an 
+`analysis_config` and an `input` object.
```
Comment (benwtrent) [inline docs/reference/ml/ml-shared.asciidoc]: this is only true if the model was created by a data frame analytics job. So, maybe this sentence should be "for example, models created by data frame analytics contain...`

File: docs/reference/ml/ml-shared.asciidoc [hunk 2]
```diff
@@ -1422,6 +1423,131 @@ The number of `partition` field values that were analyzed by the models. This
...
+
+`analytics_config`:::
```
Comment (benwtrent) [inline docs/reference/ml/ml-shared.asciidoc]: none of these `analytics*` entries should be documented. These happen to be returned in models created by an analytics process but are not guaranteed to be present.

**Comment by szabosteve** [comment]
Thanks @benwtrent!
I addressed your feedback in da529a3.

**Review by benwtrent — APPROVED** [review]

**Comment by szabosteve** [comment]
**Backport commits:**
7.x: 6cece3a
7.6: 07eda6c

**Pull request merged**